manifests/init.pp

```diff
@@ -47,6 +47,10 @@
       fail("invalid version ${package_version} requested")
     }
 
+    $aio_upgrade_required = ($is_pe == false and $package_version != undef) or
+      ($::aio_agent_version != undef and $package_version != undef and
```

Comment:
We still don't know why aio_agent_version might not be set?

Eventually we can drop the is_pe check, when we add a custom fact to this module.